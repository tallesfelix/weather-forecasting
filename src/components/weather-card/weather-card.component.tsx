import { format } from 'date-fns';
import styles from './weather-card.module.css';
import { useEffect, useState } from 'react';

const loadIcon = async (iconName: string) => {
	const icon = await import(`../../assets/icons/${iconName}.svg`);
	return icon.default;
};

type WeatherCardProps = {
	conditions: string;
	tempMin: number;
	tempMax: number;
	date: string;
	icon: string;
	variant?: 'today' | 'forecast';
}

export const WeatherCard = ({ conditions, tempMin, tempMax, date, icon, variant = 'forecast' }: WeatherCardProps) => {
	const [iconUrl, setIconUrl] = useState<string>('');

	useEffect(() => {
		loadIcon(icon).then((icon) => setIconUrl(icon));
	}, [icon]);

	return (
		<div className={variant === 'forecast' ? styles.weatherCard : styles.todaysForecasting}>
			<div>
				<img src={iconUrl} alt={icon} />
				<p>{conditions}</p>
			</div>
			<div>
				<span>min. {tempMin}°C</span>
				<span>max. {tempMax}°C</span>
				<p>{format(new Date(`${date}T01:01`), 'd/L')}</p>
			</div>
		</div>
	);
};