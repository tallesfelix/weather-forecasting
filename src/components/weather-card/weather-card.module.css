.weatherCard {
  text-align: center;
  background: '#fff';
  max-width: 100px;
  height: 100%;
  border-radius: 8px;
  border: 1px solid #ccc;
  padding: 0px 8px;
  min-height: 174px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;

  & p {
    font-weight: 600;
  }

  & span {
    font-size: 14px;
    display: block;
  }

  & img {
    width: 52px;
    height: 52px;
  }
}

.todaysForecasting {
  position: absolute;
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 10px;
  max-width: 550px;
  background: white;
  padding: 10px;
  border-radius: 8px;

  top: 50%;
  left: 50%;
  transform: translate(-50%, 100%);

  & p {
    font-weight: 600;
  }

  & img {
    width: 52px;
    height: 52px;
  }
}