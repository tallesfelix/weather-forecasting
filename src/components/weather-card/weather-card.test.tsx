import { render, screen } from '@testing-library/react';
import { WeatherCard } from './weather-card.component';

describe('WeatherCard', () => {
  test('renders WeatherCard component', () => {
    const { asFragment } = render(<WeatherCard conditions="Snow" tempMin={0} tempMax={4} date="2021-01-01" icon="snow" />);
    expect(screen.getByText(/Snow/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});