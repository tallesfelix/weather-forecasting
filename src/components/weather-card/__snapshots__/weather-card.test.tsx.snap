// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard renders WeatherCard component 1`] = `
<DocumentFragment>
  <div
    class="weatherCard"
  >
    <div>
      <img
        alt="snow"
        src=""
      />
      <p>
        Snow
      </p>
    </div>
    <div>
      <span>
        min. 0°C
      </span>
      <span>
        max. 4°C
      </span>
      <p>
        1/1
      </p>
    </div>
  </div>
</DocumentFragment>
`;
